### ISSUE TITLE
Infinite loop detected and caused by Layout/HashAlignment, Lint/BooleanSymbol, Style/HashSyntax

### ISSUE THREAD

**Issue by anykeyh** [rubocop/rubocop#14123]
I am facing an infinite loop issue when calling `rubocop -A` over this file:

https://gist.github.com/anykeyh/c9838f56066bead4383a0153d3a9bee6

--------

## Expected behavior

No infinite loop

## Actual behavior

```
1 error occurred:
Infinite loop detected in /xxxx/context.rb and caused by Layout/HashAlignment, Lint/BooleanSymbol, Style/HashSyntax
```

## Steps to reproduce the problem

Source file is here:

https://gist.github.com/anykeyh/c9838f56066bead4383a0153d3a9bee6

Rubocop config:

```yaml
inherit_from:
  - https://relaxed.ruby.style/rubocop.yml

AllCops:
  TargetRubyVersion: 3.2

Style/StringLiterals:
  Enabled: true
  EnforcedStyle: double_quotes

Style/StringLiteralsInInterpolation:
  Enabled: true
  EnforcedStyle: double_quotes

Style/AccessModifierDeclarations:
  Enabled: false

Layout/LineLength:
  Max: 120
  Enabled: true

Layout/SpaceAroundOperators:
  Enabled: true
  EnforcedStyleForExponentOperator: space
  EnforcedStyleForRationalLiterals: space

Layout/FirstMethodArgumentLineBreak:
  Enabled: true

Layout/MultilineMethodArgumentLineBreaks:
  Enabled: true

Layout/ParameterAlignment:
  Enabled: true
  EnforcedStyle: with_first_parameter

Layout/ArrayAlignment:
  Enabled: true
```

## RuboCop version

```
1.75.3 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 3.2, running on ruby 3.2.2) [x86_64-linux]
```
